File: src_local/agents/Python/Strategic/warden.py
from src_local.agents.base import BaseActivePersona
import logging
import time

logger = logging.getLogger(__name__)

class WardenPersona(BaseActivePersona):
    """
    Core: PhD in Ethics & Privacy Governance ⚖️
    Guardião da ética técnica, previne o vazamento de segredos em logs e auditorias de privacidade.
    """
    
    def __init__(self, project_root):
        super().__init__(project_root)
        self.name, self.emoji, self.role, self.stack = "Warden", "⚖️", "PhD Privacy Officer", "Python"

    def perform_audit(self) -> list:
        start_time = time.time()
        logger.info(f"[{self.name}] Analisando Privacidade e Governança (Legacy Stack)...")
        
        audit_rules = [
            {'regex': r"(?:print|logger\.\w+)\(.*?password.*?=.*?\)", 'issue': 'Vazamento Ético: Senha exposta em saída padrão ou log. Viola a privacidade PhD.', 'severity': 'critical'},
            {'regex': r"AKIA[0-9A-Z]{16}", 'issue': 'Vazamento Crítico: AWS Access Key ID detectada. Risco de exposição de infraestrutura.', 'severity': 'critical'}
        ]
        
        results = self.find_patterns(('.py',), audit_rules)
        self._log_performance(start_time, len(results))
        return results

    def _reason_about_objective(self, objective, file, content):
        if "password" in content.lower() and ("print" in content or "logger" in content):
            return f"Risco de Governança: O objetivo '${objective}' exige confiança. Em '${file}', o tratamento inseguro de segredos compromete a ética PhD."
        return f"PhD Ethics: Analisando governança de dados para {objective}. Focando em proteção de privacidade e conformidade ética."

    def self_diagnostic(self) -> dict:
        """Auto-Cura Soberana (Legacy Python)."""
        return {
            "status": "Soberano",
            "score": 100,
            "issues": []
        }

    def get_system_prompt(self):
        return f"Você é o Dr. {self.name}, PhD em Ética Digital e Guardião da Privacidade Legada."

File: src_local/agents/Rust/Strategic/warden.ts
import { BaseActivePersona } from "../../base.ts";
import type { AuditRule, StrategicFinding, AuditFinding } from "../../base.ts";
import type { ProjectContext } from "../../../core/types.ts";

/**
 * 🛡️ Dr. Warden — PhD in Rust Sovereignty & Compliance
 */
export class WardenPersona extends BaseActivePersona {
    constructor(projectRoot?: string) {
        super(projectRoot as any);
        this.name = "Warden";
        this.emoji = "🛡️";
        this.role = "PhD Strategic Guardian";
        this.phd_identity = "Sovereignty & Strategic Security (Rust)";
        this.stack = "Rust";
    }

    public override async execute(context: any): Promise<any> {
        this.setContext(context);
        const findings = await this.performAudit();
        return findings;
    }

    override getAuditRules(): { extensions: string[]; rules: AuditRule[] } {
        return {
            extensions: ['.rs'],
            rules: [
                { regex: /std::env/, issue: 'Segurança: Acesso a env vars.', severity: 'medium' }
            ]
        };
    }

    public audit(): any[] { return []; }
    public Branding(): string { return `${this.emoji} ${this.name}`; }
    public Analysis(): string { return "Security Sovereignty Analysis Complete"; }
    public test(): boolean {
        this.Branding();
        this.Analysis();
        this.audit();
        return true;
    }

    override reasonAboutObjective(objective: string, file: string, content: string | Promise<string | null>): StrategicFinding | string | null {
        return {
            file, severity: "INFO",
            issue: `PhD Warden (Rust): Analisando soberania para ${objective}.`,
            context: "analyzing sovereignty"
        };
    }

    override selfDiagnostic(): any {
        return { status: "Soberano", score: 100, issues: [], branding: this.Branding() };
    }

    override getSystemPrompt(): string {
        return `Você é o Dr. ${this.name}, PhD em soberania Rust. Status: ${this.Analysis()}`;
    }
}
